src/index.rs

```diff
@@ -387,6 +430,9 @@ impl IndexTable {
 		let offset = META_SIZE + index as usize * CHUNK_LEN;
 		// Nasty mutable pointer cast. We do ensure that all chunks that are being written are accessed
 		// through the overlay in other threads.
+
+		// [Review] most api uses '&self', seems like we could use &mut some place and remove
+		// some atomics, also not sure why we don't get write lock here.
```

Comment:
We don't want to lock all of the mapped memory and block reads for performance reasons. There needs to be synchronisation on the page level.